File: 3. Scripts/javascript/pa-shortpw2.js
const { Builder } = require('selenium-webdriver');
const assert = require('assert');
const SignupPage = require('./signup_page');
const UserPage = require('./user_page');

const timestamp = Math.floor(Date.now() / 1000);
const email = `test${timestamp}@test.me`;
const password = 'Testic';
const password2 = 'Testic';
const shortPwText = 'Šifra je kraća od 8 znakova';

describe('Polovni automobili application:', function () {
  describe('signup to the polovni automobili application', function () {
    it('confirm user use short password', async function () {
      const driver = await new Builder().forBrowser('firefox').build();

      // Go to sign-in form
      await driver.get('https://www.polovniautomobili.com/registracija');

      // Fill out form
      const signup = new SignupPage(driver);
      await signup.enterEmailAddress(email);
      await signup.enterPassword(password);
      await signup.enterPassword2(password2);
      await signup.clickRules();
      await signup.clickEasySale();
      await signup.clickEasyBuy();
      await signup.clickLogin();

      await driver.sleep(2000);

      // Confirm user use short password
      const user = new UserPage(driver);
      const shortPwBox = await user.userShortPw();
      assert.strictEqual(shortPwBox, shortPwText);

      await driver.quit();
    });
  });
});

File: 3. Scripts/pa-shortpw2.rb
require 'selenium-webdriver'
require 'rspec'
require_relative "singup_page.rb"
require_relative "user_page.rb"

timestemp = Time.now.to_i
email = "test#{timestemp}@test.me"
password = 'Testic'
password2 = 'Testic'
shortpw_text = "Šifra je kraća od 8 znakova"


# Test for registration at polovni automoboili
describe 'Polovni automobili application:' do
  describe 'signup to the polovni automobili application' do
    it 'confirm that a user can successfully signup' do
      @driver = Selenium::WebDriver.for :firefox
      # Go to sing-in form
      @driver.navigate.to 'https://www.polovniautomobili.com/registracija'

      # Fill out form
      signup = SignupPage.new(@driver)
      signup.enter_email_address(email)
      signup.enter_password(password)
      signup.enter_password2(password2)
      signup.click_rules
      signup.click_easysale
      signup.click_easybuy
      signup.click_login


      sleep 2

      # Confirm user is singed up successufly
      user = UserPage.new(@driver)
      shortpw_box = user.user_shortpw
      expect(shortpw_box).to eq(shortpw_text)
      
      @driver.quit

    end
  end
end
